src/corehost/cli/fxr/hostfxr.cpp

```diff
@@ -514,3 +565,35 @@ SHARED_API int32_t hostfxr_get_native_search_directories(const int argc, const p
     int rc = muxer.execute(_X("get-native-search-directories"), argc, argv, startup_info, buffer, buffer_size, required_buffer_size);
     return rc;
 }
+
+
+
+typedef void(*hostfxr_error_writer_fn)(const pal::char_t* message);
+
+//
+// Sets a callback which is to be used to write errors to.
+//
+// Parameters:
+//     error_writer 
+//         A callback function which will be invoked every time an error is to be reported.
+//         Or nullptr to unregister previously registered callback and return to the default behavior.
+// Return value:
+//     The previously registered callback (which is now unregistered), or nullptr if no previous callback
+//     was registered
+// 
+// The error writer is registered per-thread, so the registration is thread-local. On each thread
+// only one callback can be registered. Subsequent registrations overwrite the previous ones.
+// 
+// By default no callback is registered in which case the errors are written to stderr.
+// 
+// Each call to the error writer is sort of like writing a single line (the EOL character is omitted).
+// Multiple calls to the error writer may occure for one failure.
+//
+// If the hostfxr invokes functions in hostpolicy as part of its operation, the error writer
+// will be propagated to hostpolicy for the duration of the call. This means that errors from
+// both hostfxr and hostpolicy will be reporter through the same error writer.
+//
+SHARED_API hostfxr_error_writer_fn hostfxr_set_error_writer(hostfxr_error_writer_fn error_writer)
+{
+    return trace::set_error_writer(error_writer);
```

Comment:
Should this propagate to hostpolicy if it has already been loaded?  Or should we add a flag to allow it?  Or export a propagate function?

Reply:
The propagation will occur right before hostfxr calls into hostpolicy and will be reverted right after the call is done. The end result is that the registration seems to be dll specific, but if I call something on hostfxr which also needs hostpolicy, the call will behave as a single unit and use the same error writer for both (for the duration of the call).